tgui/packages/tgui/interfaces/ControllerOverview.tsx

```diff
@@ -0,0 +1,227 @@
+import { BooleanLike } from 'common/react';
+import { createSearch } from 'common/string';
+import { useMemo, useState } from 'react';
+
+import { useBackend } from '../backend';
+import {
+  Button,
+  Collapsible,
+  Dropdown,
+  Input,
+  Section,
+  Stack,
+} from '../components';
+import { Window } from '../layouts';
+
+type SubsystemData = {
+  name: string;
+  ref: string;
+  init_order: number;
+  last_fire: number;
+  next_fire: number;
+  can_fire: BooleanLike;
+  doesnt_fire: BooleanLike;
+  cost_ms: number;
+  tick_usage: number;
+  tick_overrun: number;
+  initialized: BooleanLike;
+  initialization_failure_message: string | undefined;
+};
+
+type ControllerData = {
+  world_time: number;
+  fast_update: BooleanLike;
+  map_cpu: number;
+  subsystems: SubsystemData[];
+};
+
+const SubsystemView = (props: { data: SubsystemData }) => {
+  const { act } = useBackend();
+  const { data } = props;
```

Comment:
prefer destructuring this here so you dont have to call data. later. general rule